rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -84,7 +84,11 @@ nested_type = '%s__%s__%s' % (field.type.pkg_name, 'msg', field.type.type)
   @(nested_type) * dest@(field.name) = ros_message->@(field.name).data;
 @[      else]@
   size_t size@(field.name) = @(field.type.array_size);
+@[        if field.type.is_upper_bound]@
+  @(nested_type) * dest@(field.name) = ros_message->@(field.name).data;
+@[        else]@
   @(nested_type) * dest@(field.name) = ros_message->@(field.name);
+@[        end if]@
```

Comment:
I think that you want to call the `__Array__init()` methods for bounded arrays.
I would suggest checking `is_upper_bound` L79 and if the array is bounded, use `min(PySequence_Size(),upper_bound)` to set the array size.

Same below